subsys/net/ip/utils.c

```diff
@@ -711,6 +711,17 @@ uint16_t net_calc_chksum_igmp(uint8_t *data, size_t len)
 }
 #endif /* CONFIG_NET_IPV4_IGMP */
 
+#if defined(CONFIG_NET_IPV4_IGMPV3)
+uint16_t net_calc_chksum_igmpv3(struct net_pkt *pkt)
```

Comment:
Can't we have just one `net_calc_chksum_igmp()` function that could handle all versions?

Reply:
The `net_calc_chksum_igmp()` is now used for both IGMPv2 and IGMPv3